packages/driver/src/cy/snapshots.ts

```diff
@@ -239,6 +239,15 @@ export const create = ($$: $Cy['$$'], state: StateFunc) => {
       return null
     }
 
+    const timestamp = performance.now() + performance.timeOrigin
+
+    // if the protocol has been enabled, our snapshot is just the name and timestamp,
+    // also make sure numTestsKeptInMemory is 0, otherwise we will want the full snapshot
+    // (the driver test's set numTestsKeptInMemory to 1 in run mode to verify the snapshots)
+    if (Cypress.config('protocolEnabled') && Cypress.config('numTestsKeptInMemory') === 0) {
```

Comment:
should we refactor the logic to a function since we use it in multiple places?

Reply:
The `if` statements are slightly different, this one is asking if we are taking protocol snapshots and the other one is asking if we are taking snapshots at all.

Reply:
> Only returning this when numTestsKeptInMemory could possible trip people up when protocol is enabled due to the fact we've shared to users to set this to 0 to help with memory issues. 

The protocol isn't enabled in open mode so this won't be an issue since execution would've ceased in `log.ts` before it ever got here.

> What is the reason we are checking on numTestsKeptInMemory=0? vs being in run mode?

As you mentioned, the main reason for checking `numTestsKeptInMemory=0` is because of the driver tests.

> Does the driver test need updated instead and/or what is the impact if we don't check for this when the numTestsKeptInMemory=1 in run mode?

The impact of not checking `numTestsKeptInMemory` is that the driver tests that verify snapshots will fail. I doubt we want to rewrite the driver tests that verify snapshots.